05week/spaceTravelToMars.js

```diff
@@ -1,51 +1,214 @@
+// console.log(mav.missionStatement());
+// console.log(hermes.crew)
+// //-------------------------------------------------------------------------------
+// //tests
+// if (typeof describe === "function") {
+//   describe("CrewMember", function () {
+//     it("should have a name, a job, a specialSkill and ship upon instantiation", function () {
+//       var crewMember1 = new CrewMember("Rick Martinez", "pilot", "chemistry");
+//       assert.equal(crewMember1.name, "Rick Martinez");
+//       assert.equal(crewMember1.job, "pilot");
+//       assert.equal(crewMember1.specialSkill, "chemistry");
+//       assert.equal(crewMember1.ship, null);
+//     });
+
+//     it("can enter a ship", function () {
+//       let mav = new Ship("Mars Ascent Vehicle", "MAV", "Ascend into low orbit");
+//       let crewMember1 = new CrewMember("Rick Martinez", "pilot", "chemistry");
+//       crewMember1.enterShip(mav);
+//       assert.equal(crewMember1.ship, mav);
+//       assert.equal(mav.crew.length, 1);
+//       assert.equal(mav.crew[0], crewMember1);
+//     });
+//   });
+
+//   describe("Ship", function () {
+//     it("should have a name, a type, an ability and an empty crew upon instantiation", function () {
+//       let mav = new Ship("Mars Ascent Vehicle", "MAV", "Ascend into low orbit");
+//       assert.equal(mav.name, "Mars Ascent Vehicle");
+//       assert.equal(mav.type, "MAV");
+//       assert.equal(mav.ability, "Ascend into low orbit");
+//       assert.equal(mav.crew.length, 0);
+//     });
+
+//     it("can return a mission statement correctly", function () {
+//       let mav = new Ship("Mars Ascent Vehicle", "MAV", "Ascend into low orbit");
+//       let crewMember1 = new CrewMember("Rick Martinez", "pilot", "chemistry");
+//       let hermes = new Ship("Hermes", "Main Ship", "Interplanetary Space Travel");
+//       let crewMember2 = new CrewMember("Commander Lewis", "commander", "geology");
+//       assert.equal(mav.missionStatement(), "Can't perform a mission yet.");
+//       assert.equal(hermes.missionStatement(), "Can't perform a mission yet.");
 
-let assert = require('assert');
+//       crewMember1.enterShip(mav);
+//       assert.equal(mav.missionStatement(), "Ascend into low orbit");
 
-let jobTypes = {
+//       crewMember2.enterShip(hermes);
+//       assert.equal(hermes.missionStatement(), "Interplanetary Space Travel");
+//     });
+//   });
+// }
+'use strict';
+//require assert
+var assert = require('assert');
+
+const jobTypes = {
   pilot: 'MAV',
   mechanic: 'Repair Ship',
   commander: 'Main Ship',
   programmer: 'Any Ship!'
 };
 
-// Your code here
+// test one
+
+class CrewMember {
+  constructor (name, job, specialSkill) {
+    this.name = name;
+    this.job = job;
+    this.specialSkill = specialSkill;
+    this.ship = null;
+    this.jobType = jobTypes[this.job];
+    // test two
+    this.enterShip = (ship) => {
+      this.ship = ship;
+      ship.crew.push(this);
```

Comment:
Your putting the entire class in the crew array.

Reply:
changed this to this.ship